pkg/topology/kademlia/internal/metrics/metrics.go

```diff
@@ -18,14 +18,50 @@ import (
 	"github.com/syndtr/goleveldb/leveldb"
 )
 
+const ewmaSmoothing = 0.1
+
 // PeerConnectionDirection represents peer connection direction.
 type PeerConnectionDirection string
 
 const (
 	PeerConnectionDirectionInbound  PeerConnectionDirection = "inbound"
 	PeerConnectionDirectionOutbound PeerConnectionDirection = "outbound"
+)
+
+// PeerReachabilityStatus represents the peer reachability status.
+type PeerReachabilityStatus int
+
+// String implements the fmt.Stringer interface.
+func (rs PeerReachabilityStatus) String() string {
+	str := [...]string{reachabilityUnknown, reachabilityPublic, reachabilityPrivate}
+	if rs < 0 || int(rs) >= len(str) {
+		return "(unrecognized)"
+	}
+	return str[rs]
+}
 
-	ewmaSmoothing = 0.1
+// ParseReachabilityStatus tries to parse reachability status from the given string.
+func ParseReachabilityStatus(s string) (PeerReachabilityStatus, error) {
+	switch s {
+	case reachabilityUnknown:
+		return PeerReachabilityStatusUnknown, nil
+	case reachabilityPublic:
+		return PeerReachabilityStatusPublic, nil
+	case reachabilityPrivate:
+		return PeerReachabilityStatusPrivate, nil
+	}
+	return -1, fmt.Errorf("unrecognized reachability status: %q", s)
+}
+
+const (
+	PeerReachabilityStatusUnknown PeerReachabilityStatus = 0 // Mirrors the network.ReachabilityUnknown.
+	PeerReachabilityStatusPublic  PeerReachabilityStatus = 1 // Mirrors the network.ReachabilityPublic.
+	PeerReachabilityStatusPrivate PeerReachabilityStatus = 2 // Mirrors the network.ReachabilityPrivate.
```

Comment:
nitpick, consider removing `Peer` from the type name and then we can repurpose the same type to indicate both the node's reachability or peer reachability